cvmfs/download.h

```diff
@@ -138,6 +145,11 @@ struct JobInfo {
     probe_hosts = false;
     head_request = false;
     follow_redirects = false;
+    pid = -1;
```

Comment:
Oh - to be clear, for now, I only disable caching specifically for handles that require a client certificate.

"Normal" handles - even non-client-certificate HTTPS - keep HTTP1.1 keepalive and SSL session reuse enabled.

I figure this way we don't adversely affect the 'normal' use case and we have minimal functionality for the secure use case.  I see creating a separate pool of curl handles as a follow-up branch (this branch already touches a dangerously large number of interfaces!).